tallycash/extension
# Arbitrum Odyssey banners

Resolves #2426
Resolves #2417

### What

Create a notification banner on the Wallet page.

This feature allows you to:
- define banner campaigns, their start and end dates, relevant chains
- fetch data from Galxe API and show latest two Odyssey campaigns
- dismiss the current banner and then show the user banner when new Odyssey campaign is available
- dismiss all banners and then change that setting in the Settings
 
![image](https://user-images.githubusercontent.com/20949277/198282191-166b11c8-27c2-4916-9253-303280fc58d6.png)
![image](https://user-images.githubusercontent.com/20949277/198282252-2cef5a3c-6033-4544-8266-359d674ea924.png)

### Testing

We currently have implemented Odyssey campaign banners - we will show 2 most recent available campaigns from Arbitrum 

To test it (for devs):
- change the [start date here](https://github.com/tallycash/extension/blob/wallet-achievements-banner/ui/components/Wallet/Banner/useBannerCampaigns.ts#L25) OR change the date on your computer (note that it can affect some apps - Github will log out, Zoom will stop working correctly)
- banner should show 2 most recent campaigns from [Galxe](https://galxe.com/arbitrum)
- if you click "Start now" it should hide the current banner but it will show a new one when there is new campaign available  (to test it [change this line](https://github.com/tallycash/extension/blob/wallet-achievements-banner/ui/components/Wallet/Banner/useArbitrumCampaigns.ts#L59) to: `const latest = achievements.slice(0, 1)`)
- if you click "X" and then "Yes, show notification"  it should hide the current banner but it will show banner when latest campaigns change 
- if you click "X" and then "No thanks" it should hide the banner and in the Settings the toggle should be "off". If you toggle it then the banner should be back. 


Latest build: [extension-builds-2455](https://github.com/tallycash/extension/suites/9088073412/artifacts/421044337) (as of Wed, 02 Nov 2022 12:41:36 GMT).
